languagetool-core/src/main/java/org/languagetool/CheckResults.java

```diff
@@ -18,45 +18,49 @@
  */
 package org.languagetool;
 
+import lombok.Getter;
 import org.jetbrains.annotations.NotNull;
 import org.languagetool.rules.RuleMatch;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * @since 5.3
  */
 public class CheckResults {
 
+  @Getter
   private List<RuleMatch> ruleMatches;
-  private List<Range> ignoredRanges;
+  @Getter
+  private final List<Range> ignoredRanges;
+  private final List<ExtendedSentenceRange> extendedSentenceRanges;
   private final List<SentenceRange> sentenceRanges = new ArrayList<>();
 
-  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges) {
-    this.ruleMatches = Objects.requireNonNull(ruleMatches);
-    this.ignoredRanges = Objects.requireNonNull(ignoredRanges);
-  }
 
-  public List<Range> getIgnoredRanges() {
-    return ignoredRanges;
+  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges) {
+    this(ruleMatches, ignoredRanges, Collections.emptyList());
   }
 
-  public List<RuleMatch> getRuleMatches() {
-    return ruleMatches;
+  public CheckResults(List<RuleMatch> ruleMatches, List<Range> ignoredRanges, List<ExtendedSentenceRange> extendedSentenceRanges) {
+    this.ruleMatches = Objects.requireNonNull(ruleMatches);
+    this.ignoredRanges = Objects.requireNonNull(ignoredRanges);
+    this.extendedSentenceRanges = Objects.requireNonNull(extendedSentenceRanges);
+    //TODO: use this later
+//    extendedSentenceRanges.forEach(extendedSentenceRange -> this.sentenceRanges.add(new SentenceRange(extendedSentenceRange.getFromPos(), extendedSentenceRange.getToPos())));
   }
 
   @NotNull
   public List<SentenceRange> getSentenceRanges() {
-    return sentenceRanges;
+    return Collections.unmodifiableList(this.sentenceRanges);
   }
+
   public void addSentenceRanges(List<SentenceRange> sentenceRanges) {
     this.sentenceRanges.addAll(sentenceRanges);
   }
-  
-  public void setIgnoredRanges(List<Range> ignoredRanges) {
-    this.ignoredRanges = Objects.requireNonNull(ignoredRanges);
+
+  public List<ExtendedSentenceRange> getExtendedSentenceRanges() {
+    return extendedSentenceRanges.stream().sorted().collect(Collectors.toList());
```

Comment:
Can we store this in sorted form and then just return the list instead of possibly sorting multiple times?